can/interfaces/socketcan/utils.py

```diff
@@ -38,7 +38,7 @@ def pack_filters(can_filters: Optional[typechecking.CanFilters] = None) -> bytes
     return struct.pack(can_filter_fmt, *filter_data)
 
 
-_PATTERN_CAN_INTERFACE = re.compile(r"v?can\d+")
+_PATTERN_CAN_INTERFACE = re.compile(r"(sl|v)?can\d+")
```

Comment:
Good idea! The slcan driver has been reworked for Linux 5.20 and will therefore show up as a 'normal' CAN interface (e.g. named 'can0') in the future.
But for the current setups the detection of 'slcan0' would be useful.
Btw. when updating the pattern it would make sense to take care of vxcan0 too, which is a special virtual CAN interface (pair) which allows to communicate over network namespace boundaries (analogue to veth), e.g. for setups in Docker containers, etc.

So it should be `re.compile(r"(sl|v|vx)?can\d+")`, right?

Reply:
@hartkopp, thanks for feedback!
Sure, I'll update the pattern and the test code.

BTW while experimenting with vcan I discovered that an interface name can be almost anything (even `foo,bar-baz` is accepted):
```
sudo ip link add dev foo,bar-baz type vcan
sudo ip link set up foo,bar-baz
```

I'm confused by the failed test.
The issue seems to be unrelated to the changes in this pull-request.
```
collecting ... ERROR: InvocationError for command 'D:\a\python-can\python-can\.tox\gh\Scripts\pytest.EXE' (exited with code 3221225477)
___________________________________ summary ___________________________________
ERROR:   gh: commands failed
```
Do I have to do something with it?

Reply:
Yes, it is a normal behaviour that you can name a network device whatever you like.
You can also rename them after creation.
When there is no name provided (e.g. by ìp link add type vcan`) the default `vcan%d` is taken from the vcan driver.

Reply:
Done.